crates/core/src/rlp/structs.rs

```diff
@@ -65,6 +64,21 @@ impl<'a> Decoder<'a> {
         };
         Ok((field, updated_self))
     }
+    /// Returns the next field without decoding it, i.e. the payload bytes including its prefix.
+    pub fn get_encoded_item(self) -> Result<(Vec<u8>, Self), RLPDecodeError> {
+        match get_item_with_prefix(self.payload) {
+            Ok((_is_list, field, rest)) => {
+                let updated_self = Self {
+                    payload: rest,
+                    ..self
+                };
+                Ok((field.to_vec(), updated_self))
+            }
+            Err(_) => Err(RLPDecodeError::Custom(
```

Comment:
Why the custom error? Can't we just propagate this one?

Reply:
Done!